### PR TITLE
🌏 Revise "softmax-regression-scratch_vn" - Phần 3

### LINKED ISSUE #921: Revise "softmax-regression-scratch_vn" - Phần 3
Phần này được dịch bởi:
* **di-mi-ta**
* **azraelzhor**

Nếu bạn đã dịch phần này, vui lòng bỏ qua việc revise.

### PR DESCRIPTION
**Pull request by duythanhvn**
Close #921

### CODE REVIEW

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 1]
```diff
@@ -387,10 +387,10 @@ Note that `updater` is general function to update the model parameters, which ac
 It can be either a wrapper of `d2l.sgd` or a Gluon trainer.
 -->
 
-Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã đọc qua cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
-Ở đây, chúng ta tái cấu trúc lại đoạn mã để giúp nó có thể được tái sử dụng. 
+Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã xem qua cách ta sẽ lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã xem qua cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
```
hồi quy tuyến tính là quá khứ rồi mà, `ta sẽ` làm gì nữa đâu.

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 2]
```diff
@@ -387,10 +387,10 @@ Note that `updater` is general function to update the model parameters, which ac
 It can be either a wrapper of `d2l.sgd` or a Gluon trainer.
 -->
 
-Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã đọc qua cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
-Ở đây, chúng ta tái cấu trúc lại đoạn mã để giúp nó có thể được tái sử dụng. 
+Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã xem qua cách ta sẽ lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
+Ở đây, chúng ta tái cấu trúc (*refactor*) lại đoạn mã để giúp nó có thể được tái sử dụng.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Ở đây, chúng ta tái cấu trúc (*refactor*) lại đoạn mã để sau này có thể tái sử dụng. 
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 3]
```diff
@@ -387,10 +387,10 @@ Note that `updater` is general function to update the model parameters, which ac
 It can be either a wrapper of `d2l.sgd` or a Gluon trainer.
 -->
 
-Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã đọc qua cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
-Ở đây, chúng ta tái cấu trúc lại đoạn mã để giúp nó có thể được tái sử dụng. 
+Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã xem qua cách ta sẽ lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
+Ở đây, chúng ta tái cấu trúc (*refactor*) lại đoạn mã để giúp nó có thể được tái sử dụng. 
 Đầu tiên, chúng ta định nghĩa một hàm để huấn luyện cho 1 epoch dữ liệu.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Đầu tiên, chúng ta định nghĩa một hàm để huấn luyện với một epoch dữ liệu. 
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 4]
```diff
@@ -567,10 +567,8 @@ What could be the problem with this implementation (hint: consider the domain of
 2. Hàm `cross_entropy` trong mục này được lập trình dựa vào định nghĩa của hàm mất mát entropy chéo. 
 Vấn đề gì có thể xảy ra với cách lập trình như vậy (gợi ý: xem xét miền của hàm log)?
 3. Bạn có thể nghĩ ra các giải pháp để giải quyết hai vấn đề trên không?
-4. Việc trả về nhãn có khả năng nhất có phải lúc nào cũng là ý tưởng tốt không? 
-Ví dụ, bạn có dùng phương pháp này cho chẩn đoán bệnh hay không?
-5. Giả sử rằng chúng ta muốn sử dụng hồi quy softmax để dự đoán từ tiếp theo dựa vào một số đặc trưng. 
-Những vấn đề gì có thể xảy ra nếu dùng một tập từ vựng lớn?
+4. Việc trả về nhãn có khả năng nhất có phải lúc nào cũng là ý tưởng tốt không? Ví dụ, bạn có dùng phương pháp này cho chẩn đoán bệnh hay không?
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
4. Việc trả về nhãn có khả năng nhất có phải lúc nào cũng là ý tưởng tốt không? Ví dụ, bạn có dùng phương pháp này để chuẩn đoán bệnh hay không?
```

Comment (duythanhvn) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Chà, chỗ này phải là **[chẩn đoán](http://dichthuatnhanhhanoi.com/10-cap-tu-hay-bi-lan-lon-trong-tieng-viet.html)** nhỉ @lkhphuc ?

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 5]
```diff
@@ -387,10 +387,10 @@ Note that `updater` is general function to update the model parameters, which ac
 It can be either a wrapper of `d2l.sgd` or a Gluon trainer.
 -->
 
-Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã đọc qua cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
-Ở đây, chúng ta tái cấu trúc lại đoạn mã để giúp nó có thể được tái sử dụng. 
-Đầu tiên, chúng ta định nghĩa một hàm để huấn luyện cho 1 epoch dữ liệu. 
-Lưu ý rằng `updater` là một hàm tổng quát để cập nhật các tham số của mô hình và sẽ nhận giá trị kích thước batch làm thông số. 
+Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã xem qua cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`.
```
Comment (tiepvupsu) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã xem cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
```

`read through` != `xem qua`

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 6]
```diff
@@ -387,10 +387,10 @@ Note that `updater` is general function to update the model parameters, which ac
...
 -->
 
-Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã đọc qua cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
-Ở đây, chúng ta tái cấu trúc lại đoạn mã để giúp nó có thể được tái sử dụng. 
-Đầu tiên, chúng ta định nghĩa một hàm để huấn luyện cho 1 epoch dữ liệu. 
-Lưu ý rằng `updater` là một hàm tổng quát để cập nhật các tham số của mô hình và sẽ nhận giá trị kích thước batch làm thông số. 
+Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã xem qua cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
+Ở đây, chúng ta tái cấu trúc (*refactor*) lại đoạn mã để sau này có thể tái sử dụng. 
+Đầu tiên, chúng ta định nghĩa một hàm để huấn luyện với một epoch dữ liệu. 
+Lưu ý rằng `updater` là một hàm tổng quát để cập nhật các tham số của mô hình và sẽ nhận kích cỡ batch làm đối số.
```
Comment (tiepvupsu) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]: Chắc phải thống nhất lại xem `batch size` là `kích thước batch` hay `kích cỡ batch`.

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 7]
```diff
@@ -416,8 +416,8 @@ Before showing the implementation of the training function, we define a utility
 Again, it aims to simplify the codes in later chapters.
 -->
 
-Trước khi xem đoạn mã thực hiện hàm huấn luyện, chúng ta định nghĩa 1 lớp phụ trợ để biểu diễn trực quan dữ liệu.
-Nhắc lại, mục đích của nó là giúp làm đơn giản hơn các đoạn mã sẽ xuất hiện trong các chương sau này. 
+Trước khi xem đoạn mã thực hiện hàm huấn luyện, ta định nghĩa một lớp phụ trợ để biểu diễn trực quan dữ liệu.
```
Comment (tiepvupsu) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Trước khi xem đoạn mã thực hiện hàm huấn luyện, ta định nghĩa một lớp phụ trợ để minh họa dữ liệu.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 8]
```diff
@@ -488,7 +488,7 @@ In practice we will want to split our data three ways into training, validation,
 Nhắc lại, chúng ta sử dụng giải thuật hạ gradient ngẫu nhiên theo minibatch để tối ưu hàm mất mát của mô hình.
 Lưu ý rằng số lượng epochs (`num_epochs`), và hệ số học (`lr`) là 2 siêu tham số được hiệu chỉnh.
```
Comment (tiepvupsu) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Lưu ý rằng số lượng epoch (`num_epochs`), và hệ số học (`lr`) là hai siêu tham số được hiệu chỉnh. 
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 9]
```diff
@@ -515,7 +515,7 @@ Given a series of images, we will compare their actual labels (first line of tex
 -->
 
 Giờ thì việc huấn luyện đã hoàn thành, mô hình của chúng ta đã sẵn sàng để phân loại các ảnh.
```
Comment (rootonchair) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Giờ thì việc huấn luyện đã hoàn thành, mô hình của chúng ta đã sẵn sàng để phân loại ảnh.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 10]
```diff
@@ -567,10 +567,8 @@ What could be the problem with this implementation (hint: consider the domain of
 2. Hàm `cross_entropy` trong mục này được lập trình dựa vào định nghĩa của hàm mất mát entropy chéo. 
 Vấn đề gì có thể xảy ra với cách lập trình như vậy (gợi ý: xem xét miền của hàm log)?
 3. Bạn có thể nghĩ ra các giải pháp để giải quyết hai vấn đề trên không?
```
Comment (ngcthuong) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
3. Bạn có thể tìm ra giải pháp cho hai vấn đề trên không?
```
